cmd/erasure-server-pool.go

```diff
@@ -874,6 +861,25 @@ func (z *erasureServerPools) DeleteObjects(ctx context.Context, bucket string, o
 		return z.serverPools[0].DeleteObjects(ctx, bucket, objects, opts)
 	}
 
+	poolObjIdxMap := map[int][]ObjectToDelete{}
```

Comment:
Now that we are here we might as well speed up fetching object locations since they are so latency sensitive.

Also issue deletes to pools concurrently.

Lightly tested, Go tests pass.

```Go
	// Fetch location of up to 10 objects concurrently.
	poolObjIdxMap := make(map[int][]ObjectToDelete, len(z.serverPools))
	origIndexMap := make(map[int][]int, len(z.serverPools))
	var mu sync.Mutex
	eg := errgroup.WithNErrs(len(objects)).WithConcurrency(10)
	ctx, cancel := eg.WithCancelOnError(ctx)
	defer cancel()
	for j, obj := range objects {
		eg.Go(func() error {
			idx, err := z.getPoolIdxExistingNoLock(ctx, bucket, obj.ObjectName)
			if isErrObjectNotFound(err) {
				derrs[j] = err
				return nil
			}
			if err != nil {
				// Unhandled errors return right here.
				return err
			}
			mu.Lock()
			poolObjIdxMap[idx] = append(poolObjIdxMap[idx], obj)
			origIndexMap[idx] = append(origIndexMap[idx], j)
			mu.Unlock()
			return nil
		}, j)
	}
	if err := eg.WaitErr(); err != nil {
		for i := range derrs {
			derrs[i] = err
		}
		return dobjects, derrs
	}

	// Delete concurrently in all server pools.
	var wg sync.WaitGroup
	wg.Add(len(z.serverPools))
	for idx, pool := range z.serverPools {
		go func(idx int, pool *erasureSets) {
			defer wg.Done()
			objs := poolObjIdxMap[idx]
			orgIndexes := origIndexMap[idx]
			deletedObjects, errs := pool.DeleteObjects(ctx, bucket, objs, opts)
			mu.Lock()
			for i, derr := range errs {
				if derr != nil {
					derrs[orgIndexes[i]] = derr
				}
				dobjects[orgIndexes[i]] = deletedObjects[i]
			}
			mu.Unlock()
		}(idx, pool)

	}
	wg.Wait()

	return dobjects, derrs
```

Reply:
Deleting is kept serial on purpose @klauspost since that causes too much random I/O

Reply:
@harshavardhana Getting the location is mostly latency and we limit it to 10. We can tweak that if we want to.

Deleting on multiple pools will never overlap in IO, since it will always be on separate disks, so there is no reason not to do it concurrently.

Reply:
okay that is good it is still serial with-in the pools.